### PR TITLE
wait_time improvements for real time booking

### PR DESCRIPTION
**Pull request by mtlmatt**
We'd like to remove the previously suggested #realtime_booking feed in favor of the existing #wait_time. These changes mainly rearranges suggestions from the feature/fix_wait_time_definition branch.

Other changes includes supporting a brand_id array to query multiple services at once, and moving to a POST request for security reasons.

### CODE REVIEW

File: reference.md
Comment (E-Baird) [inline reference.md:613]: To clarify - this PR does _not_ intend to change the (quasi) static nature of the `wait_times.json` file, correct? I'm not sure I understand the intention behind changing the description here

File: reference.md
Comment (E-Baird) [inline reference.md:665]:
```suggestion
This dynamic query provides real-time wait times for a specific location. To provide wait times to consumers, either this method or `wait_times.json` method can be used. `wait_time` allows more precise queries but requires a call on every interaction by users.
```

File: reference.md
Comment (E-Baird) [inline reference.md:667]: Not opposed to this, myself.

**Pros:**
- reduces the chance of PII being logged on anybody's systems

**Cons:**
- breaking change

```suggestion
The request must be sent as a POST request and the body must use a JSON format with the following attributes.
```

File: reference.md
Comment (E-Baird) [inline reference.md:675]: Curious as to the thought process here. Specifically, what is the intention behind enforcing at least one service brand ID?

File: reference.md
Comment (E-Baird) [inline reference.md:785]: Also not strictly opposed, again just curious as to the reasoning here. What are the value adds that you see in collapsing `wait_time` and `realtime_booking`? 

Potential downside: this will lump a _lot_ of only-kinda-sorta-related data into the `wait_time` object. In turn, maybe if we do this we should rename `wait_time` to something more generic? 

Thoughts?

File: reference.md
Comment (steph-lebl) [reply reference.md:613]: Sorry, this was a typo. No changes to wait_times.json.

File: reference.md
Comment (steph-lebl) [reply reference.md:665]: Ok

File: reference.md
Comment (steph-lebl) [reply reference.md:667]: Are we the first client to use wait_time?
If so, breaking change is not an issue. Otherwise, the proposition would be to mark GET wait_time as obsolete and to recommand using POST wait_time instead to reduces the chance of PII being logged on anybody's systems.

File: reference.md
Comment (steph-lebl) [reply reference.md:675]: Changing from a single brand_id to an array of brand_id avoid making multiple chatty http calls. For example, it allows for requesting for ["taxi-registry-standard","taxi-registry-minivan","taxi-registry-special-need"] in one request. 

Enforcing at least on service brand ID, ask the person who consumes the api to specifies the brand_id that are required instead of asking for all brand_id by default.

File: reference.md
Comment (steph-lebl) [reply reference.md:785]: estimated_wait_time, estimated_travel_time, estimated_travel_cost and estimated_travel_cost_currency are cohesive, I cannot think of a benefit to make separate http calls for this data.

The attribute booking_detail is required to support operator aggregator (ex: Montreal Taxi Registry) where estimated_wait_time and booking_detail change depending on the dispatched operator.

Equity between operators is guaranteed by always promoting the operator with the closest taxi to the customer in real time.

File: reference.md
Comment (E-Baird) [reply reference.md:675]: The switch to an array makes sense to me!

File: reference.md
Comment (E-Baird) [inline reference.md:704]: The example doesn't conform to the proposed changes (missing required `brand_id`)
```suggestion
{
  "pickup_lat": 45.60,
  "pickup_lon": -73.30,
  "brand_id": ["regular_ride", "large_ride", "shared_ride"],
}
```

File: reference.md
Comment (E-Baird) [inline reference.md:727]: Same as above - this example should be updated to use `estimated_wait_time`

File: reference.md
Comment (mtlmatt) [reply reference.md]: This part has been reverted

File: reference.md
Comment (mtlmatt) [reply reference.md]: updated

File: reference.md
Comment (mtlmatt) [reply reference.md]: updated

**Comment by E-Baird** [comment]
Discussed the proposed changes IRL with @steph-lebl and @gcamp, and we think we have a good solution. I'll create a new PR and add @steph-lebl and @mtlmatt as reviewers.

**Comment by steph-lebl** [comment]
I have changed reference.md according to our discussion with @E-Baird and @gcamp. 

We will wait for the PR approval before changing the taxi registry implementation to conform to the official spec.

File: reference.md
Comment (steph-lebl) [inline reference.md:766]: @E-Baird 
Are you ok with the encoding of array in query string?

File: reference.md
Comment (steph-lebl) [inline reference.md:752]: @E-Baird @gcamp 
You mentionned you want brand_id as optionnal, but the original spec stated "REQUIRED if more than one service brand is available.", so I left it as is. I will let Transit change the original spec.

File: reference.md
Comment (E-Baird) [reply reference.md:766]: I would prefer comma-separated values:

eg: `brand_id=regular_ride,large_ride`

File: reference.md
Comment (E-Baird) [reply reference.md:752]: Good catch. I'll discuss with people on our side and if we need to change this, we can do it later in another PR. It shouldn't affect Taxi MTL as we will make sure to always send you a value here.

File: reference.md
Comment (E-Baird) [inline reference.md:861]: Just added some extra description to make it clear why `realtime_booking` and `wait_time` both exist.

```suggestion
This dynamic query provides time/cost estimates and realtime_booking information for specific location. It can be used in situations where a provider expects to have more fine-grained or accurate booking information for a given trip, compared to what is available via other endpoints.

For example, a provider may be able to give more accurate pricing information for a specific trip, compared to the broader rules-based pricing information available from `fares.json`.
```

**Review by E-Baird — CHANGES_REQUESTED** [review]
A few small changes, but overall I think this looks good 👍

File: reference.md
Comment (steph-lebl) [reply reference.md]: Ok, see my last commit.

**Review by E-Baird — APPROVED** [review]

**Pull request merged**
